packages/platform-browser/src/hydration.ts

```diff
@@ -98,10 +98,23 @@ export function withI18nSupport(): HydrationFeature<HydrationFeatureKind.I18nSup
 }
 
 /**
- * Enables support for event replay
+ * Enables support for event replay when using clientside hydration.
+ * This feature scans all event listeners on the page and installs an event listener
+ * on the body before most JS has loaded and executed. Once this is done, the events
+ * are replayed so that relevant event listeners are executed.
```

Comment:
A proposal on how we can update docs to explain the feature (this info would be available on angular.dev), but don't describe implementation details:
```suggestion
 * Enables support for replaying user events (e.g. `click`s) that happened on a page
 * before hydration logic has completed. Once an application is hydrated, all captured events
 * are replayed and the relevant event listeners are executed.
```